diff --git a/package.json b/package.json
index e6792998475d31a9e593d73e3d9ef01a64ca8ce2..dcf140c66681496ff74980314731c3b56650295f 100644
--- a/package.json
+++ b/package.json
@@ -2,9 +2,8 @@
   "name": "@azzapp/react-native-skia-video",
   "version": "0.7.0",
   "description": "video support for react-native-skia",
-  "main": "lib/commonjs/index",
-  "module": "lib/module/index",
-  "types": "lib/typescript/src/index.d.ts",
+  "main": "src/index",
+  "module": "src/index",
   "react-native": "src/index",
   "source": "src/index",
   "files": [
diff --git a/src/exportVideoComposition.ts b/src/exportVideoComposition.ts
index d9f73c28c86da7f89a6b98f04f20b5a1970f672f..7a86189f6ee1cd74ed7bec6deb9cfa4f3c7c7234 100644
--- a/src/exportVideoComposition.ts
+++ b/src/exportVideoComposition.ts
@@ -13,13 +13,14 @@ import type {
   VideoCompositionEncoder,
   VideoCompositionFramesSyncExtractor,
 } from './types';
-import RNSkiaVideoModule from './RNSkiaVideoModule';
+import RNSkiaVideoModule from './RNSkiaVideoModule'
+
+const isAndroid = Platform.OS === 'android';
 
 const runOnNewThread = (fn: () => void) => {
   const exportRuntime = createWorkletRuntime(
     'RNSkiaVideoExportRuntime-' + performance.now()
   );
-  const isAndroid = Platform.OS === 'android';
   runOnRuntime(exportRuntime, () => {
     'worklet';
     if (isAndroid) {
@@ -30,6 +31,8 @@ const runOnNewThread = (fn: () => void) => {
   })();
 };
 
+
+
 /**
  * Exports a video composition to a video file.
  * @param videoComposition The video composition to export.
@@ -84,6 +87,11 @@ export const exportVideoComposition = async (
             height: options.height,
           });
           surface.flush();
+          if (!isAndroid && i === 0) {
+            surface.makeImageSnapshot().dispose();
+            surface.flush();
+          }
+          //@ts-expect-error not typed
           const texture = surface.getBackendTexture();
           encoder.encodeFrame(texture, currentTime);
           if (onProgress) {
