diff --git a/cjs/index.js b/cjs/index.js
index 42259d6dc48bbc0b85d27298f752b3b2625dfe5c..707b47aa5744cab5f442162c491e6ae32f171f1f 100644
--- a/cjs/index.js
+++ b/cjs/index.js
@@ -4,7 +4,7 @@ exports.useSentry = exports.defaultSkipError = void 0;
 const tslib_1 = require("tslib");
 const graphql_1 = require("graphql");
 const core_1 = require("@envelop/core");
-const Sentry = tslib_1.__importStar(require("@sentry/node"));
+const Sentry = tslib_1.__importStar(require("@sentry/nextjs"));
 exports.defaultSkipError = core_1.isOriginalGraphQLError;
 const useSentry = (options = {}) => {
     function pick(key, defaultValue) {
diff --git a/esm/index.js b/esm/index.js
index 3de73bd6bc83a2f720a54d45cf443ac5a496153b..b5aed07b0a64be54088f2fdca5bb56d6a78270ba 100644
--- a/esm/index.js
+++ b/esm/index.js
@@ -1,6 +1,6 @@
 import { Kind, print } from 'graphql';
 import { getDocumentString, handleStreamOrSingleExecutionResult, isOriginalGraphQLError, } from '@envelop/core';
-import * as Sentry from '@sentry/node';
+import * as Sentry from '@sentry/nextjs';
 export const defaultSkipError = isOriginalGraphQLError;
 export const useSentry = (options = {}) => {
     function pick(key, defaultValue) {
diff --git a/typings/index.d.cts b/typings/index.d.cts
index 5f191dd8541bbac5428d2a085ad9592bb356ad89..2832fcc024defc2caf755419d887eef1b570c0ac 100644
--- a/typings/index.d.cts
+++ b/typings/index.d.cts
@@ -1,5 +1,5 @@
 import { isOriginalGraphQLError, TypedExecutionArgs, type Plugin } from '@envelop/core';
-import * as Sentry from '@sentry/node';
+import * as Sentry from '@sentry/nextjs';
 import type { TraceparentData } from '@sentry/types';
 export type SentryPluginOptions<PluginContext extends Record<string, any>> = {
     /**
diff --git a/typings/index.d.ts b/typings/index.d.ts
index 5f191dd8541bbac5428d2a085ad9592bb356ad89..2832fcc024defc2caf755419d887eef1b570c0ac 100644
--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -1,5 +1,5 @@
 import { isOriginalGraphQLError, TypedExecutionArgs, type Plugin } from '@envelop/core';
-import * as Sentry from '@sentry/node';
+import * as Sentry from '@sentry/nextjs';
 import type { TraceparentData } from '@sentry/types';
 export type SentryPluginOptions<PluginContext extends Record<string, any>> = {
     /**
