diff --git a/cjs/index.js b/cjs/index.js
index 840e1407ccf195a4860e361d6c16873c8176d736..6f011282f23dabbf3892caa4408d338e6630f5d9 100644
--- a/cjs/index.js
+++ b/cjs/index.js
@@ -4,7 +4,7 @@ exports.useSentry = exports.defaultSkipError = void 0;
 const tslib_1 = require("tslib");
 const graphql_1 = require("graphql");
 const core_1 = require("@envelop/core");
-const Sentry = tslib_1.__importStar(require("@sentry/node"));
+const Sentry = tslib_1.__importStar(require("@sentry/nextjs"));
 exports.defaultSkipError = core_1.isOriginalGraphQLError;
 const useSentry = (options = {}) => {
     function pick(key, defaultValue) {
@@ -50,7 +50,7 @@ const useSentry = (options = {}) => {
                 forceTransaction,
                 ...traceparentData,
             }, rootSpan => {
-                rootSpan.setAttribute('document', document);
+                rootSpan?.setAttribute('document', document);
                 if (renameTransaction) {
                     Sentry.getCurrentScope().setTransactionName(transactionName);
                 }
diff --git a/esm/index.js b/esm/index.js
index 88d876f2e94a8181e0c3aeb29952337dc77f7769..fb8406f4ed0f5ceb0ca3f4abe0d26b2c4cd99483 100644
--- a/esm/index.js
+++ b/esm/index.js
@@ -1,6 +1,6 @@
 import { Kind, print } from 'graphql';
 import { getDocumentString, handleStreamOrSingleExecutionResult, isOriginalGraphQLError, } from '@envelop/core';
-import * as Sentry from '@sentry/node';
+import * as Sentry from '@sentry/nextjs';
 export const defaultSkipError = isOriginalGraphQLError;
 export const useSentry = (options = {}) => {
     function pick(key, defaultValue) {
@@ -46,7 +46,7 @@ export const useSentry = (options = {}) => {
                 forceTransaction,
                 ...traceparentData,
             }, rootSpan => {
-                rootSpan.setAttribute('document', document);
+                rootSpan?.setAttribute('document', document);
                 if (renameTransaction) {
                     Sentry.getCurrentScope().setTransactionName(transactionName);
                 }
diff --git a/typings/index.d.cts b/typings/index.d.cts
index a5b5560dbedbe53cb4132fc2c95af78f203139c4..7c8958b5792df090b455883674a687defdf0ea92 100644
--- a/typings/index.d.cts
+++ b/typings/index.d.cts
@@ -1,5 +1,5 @@
 import { isOriginalGraphQLError, TypedExecutionArgs, type Plugin } from '@envelop/core';
-import * as Sentry from '@sentry/node';
+import * as Sentry from '@sentry/nextjs';
 import type { TraceparentData } from '@sentry/types';
 export type SentryPluginOptions<PluginContext extends Record<string, any>> = {
     /**
diff --git a/typings/index.d.ts b/typings/index.d.ts
index a5b5560dbedbe53cb4132fc2c95af78f203139c4..7c8958b5792df090b455883674a687defdf0ea92 100644
--- a/typings/index.d.ts
+++ b/typings/index.d.ts
@@ -1,5 +1,5 @@
 import { isOriginalGraphQLError, TypedExecutionArgs, type Plugin } from '@envelop/core';
-import * as Sentry from '@sentry/node';
+import * as Sentry from '@sentry/nextjs';
 import type { TraceparentData } from '@sentry/types';
 export type SentryPluginOptions<PluginContext extends Record<string, any>> = {
     /**
