diff --git a/cpp/rnskia/RNSkDomView.h b/cpp/rnskia/RNSkDomView.h
index 27bb74720fdec62871d0639e3a2cd12479612a7a..2a8466cdd36d1a39c872e61eeb4aef8e7d0109b6 100644
--- a/cpp/rnskia/RNSkDomView.h
+++ b/cpp/rnskia/RNSkDomView.h
@@ -10,6 +10,7 @@
 #include <jsi/jsi.h>
 
 #include "JsiValueWrapper.h"
+#include "JsiSkCanvas.h"
 #include "RNSkView.h"
 
 #include "JsiDomRenderNode.h"
@@ -136,5 +137,20 @@ public:
       }
     }
   }
+
+  jsi::Value callJsiMethod(jsi::Runtime &runtime,
+                           const std::string &name,
+                           const jsi::Value *arguments, size_t count) override {
+    if (name == "renderToCanvas") {
+      auto callback = std::make_shared<jsi::Function>(
+          arguments[0].asObject(runtime).asFunction(runtime));
+      getCanvasProvider()->renderToCanvas([=, &runtime](SkCanvas *canvas) {
+        auto jsiCanvas = jsi::Object::createFromHostObject(
+            runtime, std::make_shared<JsiSkCanvas>(getPlatformContext(), canvas));
+        callback->call(runtime, jsiCanvas);
+      });
+    }
+    return jsi::Value::undefined();
+  }
 };
 } // namespace RNSkia
diff --git a/cpp/rnskia/RNSkJsiViewApi.h b/cpp/rnskia/RNSkJsiViewApi.h
index c1f920e1ab209c023cacc445d88a6a823d13a6db..f33f31b169f3b4f4cf136344c91fff8e77bbdff6 100644
--- a/cpp/rnskia/RNSkJsiViewApi.h
+++ b/cpp/rnskia/RNSkJsiViewApi.h
@@ -189,7 +189,54 @@ public:
         });
   }
 
+  /**
+   Calls a custom command / method on a view by the view id.
+   */
+  JSI_HOST_FUNCTION(callJsiMethod) {
+    if (count < 2) {
+      _platformContext->raiseError(
+          std::string("callCustomCommand: Expected at least 2 arguments, got " +
+                      std::to_string(count) + "."));
+
+      return jsi::Value::undefined();
+    }
+
+    if (!arguments[0].isNumber()) {
+      _platformContext->raiseError(
+          "callCustomCommand: First argument must be a number");
+
+      return jsi::Value::undefined();
+    }
+
+    if (!arguments[1].isString()) {
+      _platformContext->raiseError("callCustomCommand: Second argument must be "
+                                   "the name of the action to call.");
+
+      return jsi::Value::undefined();
+    }
+
+    auto nativeId = arguments[0].asNumber();
+    auto action = arguments[1].asString(runtime).utf8(runtime);
+
+    auto info = getEnsuredViewInfo(nativeId);
+
+    if (info->view == nullptr) {
+      throw jsi::JSError(
+          runtime, std::string("callCustomCommand: Could not call action " +
+                               action + " on view - view not ready.")
+                       .c_str());
+
+      return jsi::Value::undefined();
+    }
+
+    // Get arguments
+    size_t paramsCount = count - 2;
+    const jsi::Value *params = paramsCount > 0 ? &arguments[2] : nullptr;
+    return info->view->callJsiMethod(runtime, action, params, paramsCount);
+  }
+
   JSI_EXPORT_FUNCTIONS(JSI_EXPORT_FUNC(RNSkJsiViewApi, setJsiProperty),
+                       JSI_EXPORT_FUNC(RNSkJsiViewApi, callJsiMethod),
                        JSI_EXPORT_FUNC(RNSkJsiViewApi, requestRedraw),
                        JSI_EXPORT_FUNC(RNSkJsiViewApi, makeImageSnapshotAsync),
                        JSI_EXPORT_FUNC(RNSkJsiViewApi, makeImageSnapshot))
diff --git a/cpp/rnskia/RNSkView.h b/cpp/rnskia/RNSkView.h
index 90c46f4a4f8e45722638f2a6371006e76c920cd8..beb07f038ef75d8065981d254314c36322263c43 100644
--- a/cpp/rnskia/RNSkView.h
+++ b/cpp/rnskia/RNSkView.h
@@ -173,6 +173,17 @@ public:
     // Nothing here...
   }
 
+  /**
+   Calls a custom action.
+   */
+  virtual jsi::Value callJsiMethod(jsi::Runtime &runtime,
+                                   const std::string &name,
+                                   const jsi::Value *arguments, size_t count) {
+    throw std::runtime_error(
+        "The base Skia View does not support any commands. Command " + name +
+        " not found.");
+  }
+
   /**
    * Repaints the Skia view using the underlying context and the drawcallback.
    * This method schedules a draw request that will be run on the correct
diff --git a/src/views/SkiaDomView.tsx b/src/views/SkiaDomView.tsx
index 3f740e42e8dc16cc14ffb66068477e254c0cc1bd..b501f68819bbcc88b4eec16c4fe1ccc1f209322a 100644
--- a/src/views/SkiaDomView.tsx
+++ b/src/views/SkiaDomView.tsx
@@ -107,7 +107,8 @@ const assertSkiaViewApi = () => {
     SkiaViewApi === null ||
     SkiaViewApi.setJsiProperty === null ||
     SkiaViewApi.requestRedraw === null ||
-    SkiaViewApi.makeImageSnapshot === null
+    SkiaViewApi.makeImageSnapshot === null ||
+    SkiaViewApi.callJsiMethod === null 
   ) {
     throw Error("Skia View Api was not found.");
   }
diff --git a/src/views/types.ts b/src/views/types.ts
index 022c941d9052630769ee63d60a7dfcd9dbb87f48..bcf20c121fee96fb0c0af15f772d94f8533956b3 100644
--- a/src/views/types.ts
+++ b/src/views/types.ts
@@ -51,6 +51,11 @@ export type TouchHandler = (touchInfo: Array<Array<TouchInfo>>) => void;
 export interface ISkiaViewApi {
   setJsiProperty: <T>(nativeId: number, name: string, value: T) => void;
   requestRedraw: (nativeId: number) => void;
+  callJsiMethod: <T extends Array<unknown>>(
+              nativeId: number,
+              name: string,
+              ...args: T
+            ) => void;
   makeImageSnapshot: (nativeId: number, rect?: SkRect) => SkImage;
   makeImageSnapshotAsync: (nativeId: number, rect?: SkRect) => Promise<SkImage>;
 }
