diff --git a/cpp/api/JsiSkImageFilterFactory.h b/cpp/api/JsiSkImageFilterFactory.h
index 3488401b04d08e4f4b7b0ea7624b2aa97a920482..666432de51b5ead261e76bf80a7004ffdc33a019 100644
--- a/cpp/api/JsiSkImageFilterFactory.h
+++ b/cpp/api/JsiSkImageFilterFactory.h
@@ -37,6 +37,77 @@ public:
                                           imageFilter)));
   }
 
+
+  JSI_HOST_FUNCTION(MakeImage) {
+    auto image = JsiSkImage::fromValue(runtime, arguments[0]);
+    auto sampling = SkSamplingOptions(SkFilterMode::kLinear, SkMipmapMode::kNone);
+    return jsi::Object::createFromHostObject(
+        runtime, std::make_shared<JsiSkImageFilter>(
+                     getContext(), SkImageFilters::Image(std::move(image), sampling)));
+  }
+
+  JSI_HOST_FUNCTION(MakeMatrixTransform) {
+    auto matrix = JsiSkMatrix::fromValue(runtime, arguments[0]);
+    auto sampling = SkSamplingOptions(SkFilterMode::kLinear, SkMipmapMode::kNone);
+    sk_sp<SkImageFilter> input;
+    if (!arguments[1].isNull()) {
+      input = JsiSkImageFilter::fromValue(runtime, arguments[1]);
+    }
+    return jsi::Object::createFromHostObject(
+        runtime, std::make_shared<JsiSkImageFilter>(
+                     getContext(), SkImageFilters::MatrixTransform(
+                                       *matrix, sampling, std::move(input))));
+  }
+  
+  JSI_HOST_FUNCTION(MakeCrop) {
+    auto rect = JsiSkRect::fromValue(runtime, arguments[0]);
+    sk_sp<SkImageFilter> input;
+    if (!arguments[1].isNull()) {
+      input = JsiSkImageFilter::fromValue(runtime, arguments[1]);
+    }
+    
+    return jsi::Object::createFromHostObject(
+        runtime, std::make_shared<JsiSkImageFilter>(
+                     getContext(), SkImageFilters::Crop(
+                                       *rect, SkTileMode::kDecal, std::move(input))));
+  }
+
+  JSI_HOST_FUNCTION(MakeRuntimeShaderWithChildren) {
+    auto rtb = JsiSkRuntimeShaderBuilder::fromValue(runtime, arguments[0]);
+
+    std::vector<std::string> childNames;
+    auto childNamesJS = arguments[1].asObject(runtime).asArray(runtime);
+    size_t length = childNamesJS.size(runtime);
+    for (size_t i = 0; i < length; ++i) {
+      auto element = childNamesJS.getValueAtIndex(runtime, i);
+      childNames.push_back(element.asString(runtime).utf8(runtime).c_str());
+    }
+    std::vector<std::string_view> childNamesStringView;
+    childNamesStringView.reserve(childNames.size());
+    for (const auto& name : childNames) {
+        childNamesStringView.push_back(std::string_view(name));
+    }
+    
+    std::vector<sk_sp<SkImageFilter>> inputs;
+    auto inputsJS = arguments[2].asObject(runtime).asArray(runtime);
+    if (inputsJS.size(runtime) != length) {
+      return jsi::Value::null();
+    }
+    for (size_t i = 0; i < length; ++i) {
+      auto element = inputsJS.getValueAtIndex(runtime, i);
+      if (element.isNull()) {
+        inputs.push_back(nullptr);
+      } else {
+        inputs.push_back(JsiSkImageFilter::fromValue(runtime, element));
+      }
+    }
+    return jsi::Object::createFromHostObject(
+        runtime, std::make_shared<JsiSkImageFilter>(
+                     getContext(), SkImageFilters::RuntimeShader(
+                                       *rtb, childNamesStringView.data(), inputs.data(), length)));
+  }
+  
+
   JSI_HOST_FUNCTION(MakeColorFilter) {
     auto cf = JsiSkColorFilter::fromValue(runtime, arguments[0]);
     sk_sp<SkImageFilter> input;
@@ -223,6 +294,9 @@ public:
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeBlur),
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeOffset),
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeColorFilter),
+      JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeCrop),
+      JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeMatrixTransform),
+      JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeImage),
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeShader),
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeDisplacementMap),
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeCompose),
@@ -231,7 +305,8 @@ public:
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeBlend),
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeDropShadow),
       JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeDropShadowOnly),
-      JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeRuntimeShader))
+      JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeRuntimeShader),
+      JSI_EXPORT_FUNC(JsiSkImageFilterFactory, MakeRuntimeShaderWithChildren))
 
   explicit JsiSkImageFilterFactory(std::shared_ptr<RNSkPlatformContext> context)
       : JsiSkHostObject(std::move(context)) {}
diff --git a/lib/module/skia/types/ImageFilter/ImageFilterFactory.d.ts b/lib/module/skia/types/ImageFilter/ImageFilterFactory.d.ts
index 3ccb1e20cd143d0d8f64970f000705e72f2485db..1718bb38b5a8a0f9f7c53d972d31243db214195c 100644
--- a/lib/module/skia/types/ImageFilter/ImageFilterFactory.d.ts
+++ b/lib/module/skia/types/ImageFilter/ImageFilterFactory.d.ts
@@ -1,5 +1,7 @@
 import type { SkColor } from "../Color";
 import type { SkColorFilter } from "../ColorFilter/ColorFilter";
+import { SkImage } from "../Image/Image";
+import { SkMatrix } from "../Matrix";
 import type { BlendMode } from "../Paint";
 import type { SkRect } from "../Rect";
 import type { SkRuntimeShaderBuilder } from "../RuntimeEffect";
@@ -127,4 +129,11 @@ export interface ImageFilterFactory {
      *                         shader. If null the implicit source image is used instead
      */
     MakeRuntimeShader: (builder: SkRuntimeShaderBuilder, childShaderName: string | null, input: SkImageFilter | null) => SkImageFilter;
+
+
+
+    MakeCrop(rect: SkRect, input: SkImageFilter | null): SkImageFilter;
+    MakeMatrixTransform(matrix: SkMatrix, filter: SkImageFilter | null): SkImageFilter;
+    MakeImage(image: SkImage): SkImageFilter;
+    MakeRuntimeShaderWithChildren: (builder: SkRuntimeShaderBuilder, childNames: string[], children: (SkImageFilter | null)[]) => SkImageFilter;
 }
diff --git a/lib/typescript/src/skia/types/ImageFilter/ImageFilterFactory.d.ts b/lib/typescript/src/skia/types/ImageFilter/ImageFilterFactory.d.ts
index 3ccb1e20cd143d0d8f64970f000705e72f2485db..2f45acaeb4d7707b6711316435d08038bcdc91f4 100644
--- a/lib/typescript/src/skia/types/ImageFilter/ImageFilterFactory.d.ts
+++ b/lib/typescript/src/skia/types/ImageFilter/ImageFilterFactory.d.ts
@@ -2,6 +2,8 @@ import type { SkColor } from "../Color";
 import type { SkColorFilter } from "../ColorFilter/ColorFilter";
 import type { BlendMode } from "../Paint";
 import type { SkRect } from "../Rect";
+import { SkImage } from "../Image/Image";
+import { SkMatrix } from "../Matrix";
 import type { SkRuntimeShaderBuilder } from "../RuntimeEffect";
 import type { SkShader } from "../Shader";
 import type { SkImageFilter, TileMode } from "./ImageFilter";
@@ -127,4 +129,10 @@ export interface ImageFilterFactory {
      *                         shader. If null the implicit source image is used instead
      */
     MakeRuntimeShader: (builder: SkRuntimeShaderBuilder, childShaderName: string | null, input: SkImageFilter | null) => SkImageFilter;
+
+
+    MakeCrop(rect: SkRect, input: SkImageFilter | null): SkImageFilter;
+    MakeMatrixTransform(matrix: SkMatrix, filter: SkImageFilter | null): SkImageFilter;
+    MakeImage(image: SkImage): SkImageFilter;
+    MakeRuntimeShaderWithChildren: (builder: SkRuntimeShaderBuilder, childNames: string[], children: (SkImageFilter | null)[]) => SkImageFilter;
 }
diff --git a/src/skia/types/ImageFilter/ImageFilterFactory.ts b/src/skia/types/ImageFilter/ImageFilterFactory.ts
index ee691877c50d7f8e8f6f7e7fe0657ae7f59cd9bd..19e55f61ba9ec51649db350f2acc7dc45aabcd71 100644
--- a/src/skia/types/ImageFilter/ImageFilterFactory.ts
+++ b/src/skia/types/ImageFilter/ImageFilterFactory.ts
@@ -3,6 +3,8 @@ import type { SkColorFilter } from "../ColorFilter/ColorFilter";
 import type { BlendMode } from "../Paint";
 import type { SkRect } from "../Rect";
 import type { SkRuntimeShaderBuilder } from "../RuntimeEffect";
+import { SkImage } from "../Image/Image";
+import { SkMatrix } from "../Matrix";
 import type { SkShader } from "../Shader";
 
 import type { SkImageFilter, TileMode } from "./ImageFilter";
@@ -189,4 +191,9 @@ export interface ImageFilterFactory {
     childShaderName: string | null,
     input: SkImageFilter | null
   ) => SkImageFilter;
+
+  MakeCrop(rect: SkRect, input: SkImageFilter | null): SkImageFilter;
+  MakeMatrixTransform(matrix: SkMatrix, filter: SkImageFilter | null): SkImageFilter;
+  MakeImage(image: SkImage): SkImageFilter;
+  MakeRuntimeShaderWithChildren: (builder: SkRuntimeShaderBuilder, childNames: string[], children: (SkImageFilter | null)[]) => SkImageFilter;
 }
