diff --git a/cpp/rnskia/RNSkDomView.h b/cpp/rnskia/RNSkDomView.h
index 27bb74720fdec62871d0639e3a2cd12479612a7a..d8e5229e38c44dbfc96038a3c09b9f541be36ee3 100644
--- a/cpp/rnskia/RNSkDomView.h
+++ b/cpp/rnskia/RNSkDomView.h
@@ -10,6 +10,7 @@
 #include <jsi/jsi.h>
 
 #include "JsiValueWrapper.h"
+#include "JsiSkCanvas.h"
 #include "RNSkView.h"
 
 #include "JsiDomRenderNode.h"
@@ -32,7 +33,6 @@ class SkRect;
 class SkImage;
 
 namespace RNSkia {
-class JsiSkCanvas;
 namespace jsi = facebook::jsi;
 
 class RNSkDomRenderer : public RNSkRenderer,
@@ -136,5 +136,20 @@ public:
       }
     }
   }
+
+  jsi::Value callJsiMethod(jsi::Runtime &runtime,
+                           const std::string &name,
+                           const jsi::Value *arguments, size_t count) override {
+    if (name == "renderToCanvas") {
+      auto callback = std::make_shared<jsi::Function>(
+          arguments[0].asObject(runtime).asFunction(runtime));
+      getCanvasProvider()->renderToCanvas([=, &runtime](SkCanvas *canvas) {
+        auto jsiCanvas = jsi::Object::createFromHostObject(
+            runtime, std::make_shared<JsiSkCanvas>(getPlatformContext(), canvas));
+        callback->call(runtime, jsiCanvas);
+      });
+    }
+    return jsi::Value::undefined();
+  }
 };
 } // namespace RNSkia
