diff --git a/ios/Decoding.swift b/ios/Decoding.swift
index 2f6f500041a80fa86d0a7398896949c4370ec39c..b0f3c68540c2ce0abc00ce212109b2cde1ac7c56 100644
--- a/ios/Decoding.swift
+++ b/ios/Decoding.swift
@@ -38,7 +38,7 @@ func decodeSocialProfiles(_ input: [SocialProfile]?) -> [CNLabeledValue<CNSocial
     let username = item.username
     let userId = item.userId
     let service = item.service
-    let profile = CNSocialProfile(urlString: urlString?.path, username: username, userIdentifier: userId, service: service)
+    let profile = CNSocialProfile(urlString: urlString?.absoluteString, username: username, userIdentifier: userId, service: service)
     output.append(CNLabeledValue(label: label, value: profile))
   }
   return output
