diff --git a/ios/Decoding.swift b/ios/Decoding.swift
index b8ead58605830ce77801c3d957924b495777f0be..38c5d9a9a978384c1090dbd2e4b7f4807db6ea01 100644
--- a/ios/Decoding.swift
+++ b/ios/Decoding.swift
@@ -38,7 +38,7 @@ func decodeSocialProfiles(_ input: [SocialProfile]?) -> [CNLabeledValue<CNSocial
     let username = item.username
     let userId = item.userId
     let service = item.service
-    let profile = CNSocialProfile(urlString: urlString?.path, username: username, userIdentifier: userId, service: service)
+    let profile = CNSocialProfile(urlString: urlString?.absoluteString, username: username, userIdentifier: userId, service: service)
     output.append(CNLabeledValue(label: label, value: profile))
   }
   return output
