diff --git a/ios/RNSConvert.mm b/ios/RNSConvert.mm
index 2acf65554b8e5728adacc1d4c081c29e76bde954..b7b780a58324433eb3fce6c48222d2e2df8f63b5 100644
--- a/ios/RNSConvert.mm
+++ b/ios/RNSConvert.mm
@@ -70,6 +70,8 @@ + (RNSScreenStackAnimation)RNSScreenStackAnimationFromCppEquivalent:(react::RNSS
       return RNSScreenStackAnimationSlideFromBottom;
     case react::RNSScreenStackAnimation::Fade_from_bottom:
       return RNSScreenStackAnimationFadeFromBottom;
+    case facebook::react::RNSScreenStackAnimation::Custom:
+      return RNSScreenStackAnimationCustom;
   }
 }
 
diff --git a/ios/RNSEnums.h b/ios/RNSEnums.h
index ff2560b3a31eac74b4f5f41bab274416360b4975..de6371983b6687e2af066b66a5a3e9e244885538 100644
--- a/ios/RNSEnums.h
+++ b/ios/RNSEnums.h
@@ -17,6 +17,7 @@ typedef NS_ENUM(NSInteger, RNSScreenStackAnimation) {
   RNSScreenStackAnimationSlideFromBottom,
   RNSScreenStackAnimationSimplePush,
   RNSScreenStackAnimationSlideFromLeft,
+  RNSScreenStackAnimationCustom,
 };
 
 typedef NS_ENUM(NSInteger, RNSScreenReplaceAnimation) {
diff --git a/ios/RNSScreen.h b/ios/RNSScreen.h
index bbdedcdb9a4579377545cf011fa8f5d87d0ec05f..b582d4fa34a91f0303d358b938d71e43f5961607 100644
--- a/ios/RNSScreen.h
+++ b/ios/RNSScreen.h
@@ -63,6 +63,7 @@ namespace react = facebook::react;
 @property (nonatomic) BOOL hasHomeIndicatorHiddenSet;
 @property (nonatomic) BOOL hasOrientationSet;
 @property (nonatomic) RNSScreenStackAnimation stackAnimation;
+@property (nonatomic, copy) NSDictionary* stackAnimationOptions;
 @property (nonatomic) RNSScreenStackPresentation stackPresentation;
 @property (nonatomic) RNSScreenSwipeDirection swipeDirection;
 @property (nonatomic) RNSScreenReplaceAnimation replaceAnimation;
diff --git a/ios/RNSScreen.mm b/ios/RNSScreen.mm
index ff804e07726dcaf3dddac0134c9863844fd4da58..eda0188dc29b53a599521ee4b615a86e866f69b3 100644
--- a/ios/RNSScreen.mm
+++ b/ios/RNSScreen.mm
@@ -198,6 +198,7 @@ - (void)setStackAnimation:(RNSScreenStackAnimation)stackAnimation
     case RNSScreenStackAnimationSlideFromBottom:
     case RNSScreenStackAnimationFadeFromBottom:
     case RNSScreenStackAnimationSlideFromLeft:
+    case RNSScreenStackAnimationCustom:
       // Default
       break;
   }
@@ -1440,6 +1441,7 @@ @implementation RNSScreenManager
 RCT_EXPORT_VIEW_PROPERTY(preventNativeDismiss, BOOL)
 RCT_EXPORT_VIEW_PROPERTY(replaceAnimation, RNSScreenReplaceAnimation)
 RCT_EXPORT_VIEW_PROPERTY(stackPresentation, RNSScreenStackPresentation)
+RCT_EXPORT_VIEW_PROPERTY(stackAnimationOptions, NSDictionary)
 RCT_EXPORT_VIEW_PROPERTY(stackAnimation, RNSScreenStackAnimation)
 RCT_EXPORT_VIEW_PROPERTY(swipeDirection, RNSScreenSwipeDirection)
 RCT_EXPORT_VIEW_PROPERTY(transitionDuration, NSNumber)
@@ -1537,6 +1539,7 @@ @implementation RCTConvert (RNSScreen)
       @"slide_from_bottom" : @(RNSScreenStackAnimationSlideFromBottom),
       @"slide_from_right" : @(RNSScreenStackAnimationDefault),
       @"slide_from_left" : @(RNSScreenStackAnimationSlideFromLeft),
+      @"custom" : @(RNSScreenStackAnimationCustom),
       @"ios" : @(RNSScreenStackAnimationDefault),
     }),
     RNSScreenStackAnimationDefault,
diff --git a/ios/RNSScreenStackAnimator.h b/ios/RNSScreenStackAnimator.h
index f96eae77566c46367b3d8323669d2295f85f3dbf..415b0d1a471e208c21cba5ea6e0043d9f2452b1c 100644
--- a/ios/RNSScreenStackAnimator.h
+++ b/ios/RNSScreenStackAnimator.h
@@ -1,8 +1,20 @@
 #import "RNSScreen.h"
 
+@protocol RNSScreenCustomStackAnimator <NSObject>
+
+- (void)animateCustomWithTransitionContext:(id<UIViewControllerContextTransitioning>)transitionContext
+                                      toVC:(UIViewController *)toViewController
+                                    fromVC:(UIViewController *)fromViewController
+                              forOperation:(UINavigationControllerOperation)operation
+                               andDuration:(NSTimeInterval)duration
+                               withOptions:(NSDictionary *)options;
+
+@end
+
 @interface RNSScreenStackAnimator : NSObject <UIViewControllerAnimatedTransitioning>
 
 - (instancetype)initWithOperation:(UINavigationControllerOperation)operation;
 + (BOOL)isCustomAnimation:(RNSScreenStackAnimation)animation;
++ (void)registerCustomAnimator:(id<RNSScreenCustomStackAnimator>)animator forName:(NSString *)name;
 
 @end
diff --git a/ios/RNSScreenStackAnimator.mm b/ios/RNSScreenStackAnimator.mm
index e31d20d4eaa6aa4627cc95e1d0e83c90ab460620..f22dc95b8d7bc062fd41a0d026b1e9c067d02ce3 100644
--- a/ios/RNSScreenStackAnimator.mm
+++ b/ios/RNSScreenStackAnimator.mm
@@ -3,6 +3,7 @@
 
 #import "RNSScreen.h"
 
+
 // proportions to default transition duration
 static const float RNSSlideOpenTransitionDurationProportion = 1;
 static const float RNSFadeOpenTransitionDurationProportion = 0.2 / 0.35;
@@ -400,9 +401,56 @@ - (void)animateTransitionWithStackAnimation:(RNSScreenStackAnimation)animation
   } else if (animation == RNSScreenStackAnimationFadeFromBottom) {
     [self animateFadeFromBottomWithTransitionContext:transitionContext toVC:toVC fromVC:fromVC];
     return;
-  }
+  } else if (animation == RNSScreenStackAnimationCustom) {
+    [self animateCustomWithTransitionContext:transitionContext toVC:toVC fromVC:fromVC];
+    return;
+   }
   // simple_push is the default custom animation
   [self animateSimplePushWithTransitionContext:transitionContext toVC:toVC fromVC:fromVC];
 }
 
+
+static NSMutableDictionary<NSString *, id<RNSScreenCustomStackAnimator>> *customAnimators;
++ (void)registerCustomAnimator:(id<RNSScreenCustomStackAnimator>)animator forName:(NSString *)name {
+  if (customAnimators == nil) {
+    customAnimators = [[NSMutableDictionary alloc] init];
+  }
+  customAnimators[name] = animator;
+}
+
+- (void)animateCustomWithTransitionContext:(id<UIViewControllerContextTransitioning>)transitionContext
+                                      toVC:(UIViewController *)toViewController
+                                    fromVC:(UIViewController *)fromViewController
+{
+  RNSScreenView *screen;
+  if (_operation == UINavigationControllerOperationPush) {
+    screen = ((RNSScreen *)toViewController).screenView;
+  } else if (_operation == UINavigationControllerOperationPop) {
+    screen = ((RNSScreen *)fromViewController).screenView;
+  }
+  if (screen == nil || screen.stackAnimationOptions == nil) {
+    [self animateSimplePushWithTransitionContext:transitionContext
+                                            toVC:toViewController
+                                          fromVC:fromViewController];
+    return;
+  }
+  NSDictionary *options = screen.stackAnimationOptions;
+  NSString *animatorName = options[@"animator"];
+  id<RNSScreenCustomStackAnimator> animator = customAnimators[animatorName];
+  if (animator == nil) {
+    [self animateSimplePushWithTransitionContext:transitionContext
+                                            toVC:toViewController
+                                          fromVC:fromViewController];
+    return;
+  }
+  [animator animateCustomWithTransitionContext:transitionContext
+                                          toVC:toViewController
+                                        fromVC:fromViewController
+                                  forOperation:_operation
+                                   andDuration:[self transitionDuration:transitionContext]
+                                   withOptions:options];
+}
+
+
+
 @end
diff --git a/lib/typescript/fabric/ScreenNativeComponent.d.ts b/lib/typescript/fabric/ScreenNativeComponent.d.ts
index 17d90bea851f506b340d2feb40243b9a040667db..f5a525e124042c8dbc8c79caf489c09bb9a91c5e 100644
--- a/lib/typescript/fabric/ScreenNativeComponent.d.ts
+++ b/lib/typescript/fabric/ScreenNativeComponent.d.ts
@@ -20,7 +20,7 @@ type GestureResponseDistanceType = Readonly<{
     bottom: Float;
 }>;
 type StackPresentation = 'push' | 'modal' | 'transparentModal' | 'fullScreenModal' | 'formSheet' | 'containedModal' | 'containedTransparentModal';
-type StackAnimation = 'default' | 'flip' | 'simple_push' | 'none' | 'fade' | 'slide_from_right' | 'slide_from_left' | 'slide_from_bottom' | 'fade_from_bottom' | 'ios';
+type StackAnimation = 'default' | 'flip' | 'simple_push' | 'none' | 'fade' | 'slide_from_right' | 'slide_from_left' | 'slide_from_bottom' | 'fade_from_bottom' | 'ios' | 'custom';
 type SwipeDirection = 'vertical' | 'horizontal';
 type ReplaceAnimation = 'pop' | 'push';
 type SheetDetentTypes = 'large' | 'medium' | 'all';
@@ -63,6 +63,10 @@ export interface NativeProps extends ViewProps {
     navigationBarTranslucent?: boolean;
     navigationBarHidden?: boolean;
     nativeBackButtonDismissalEnabled?: boolean;
+    /**
+     * Custom animator options
+     */
+    stackAnimationOptions?: { animator: string } & Record<string, any>
 }
 declare const _default: import("react-native/Libraries/Utilities/codegenNativeComponent").NativeComponentType<NativeProps>;
 export default _default;
