diff --git a/node_modules/@react-native-camera-roll/camera-roll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java b/node_modules/@react-native-camera-roll/camera-roll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
index 70739fd..98718d4 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
+++ b/node_modules/@react-native-camera-roll/camera-roll/android/src/main/java/com/reactnativecommunity/cameraroll/CameraRollModule.java
@@ -95,6 +95,7 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
           MediaStore.MediaColumns.WIDTH,
           MediaStore.MediaColumns.HEIGHT,
           MediaStore.MediaColumns.SIZE,
+          MediaStore.MediaColumns.ORIENTATION,
           MediaStore.MediaColumns.DATA
   };
 
@@ -569,6 +570,7 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
     int heightIndex = media.getColumnIndex(MediaStore.MediaColumns.HEIGHT);
     int sizeIndex = media.getColumnIndex(MediaStore.MediaColumns.SIZE);
     int dataIndex = media.getColumnIndex(MediaStore.MediaColumns.DATA);
+    int orientationIndex = media.getColumnIndex(MediaStore.MediaColumns.ORIENTATION);
 
     boolean includeLocation = include.contains(INCLUDE_LOCATION);
     boolean includeFilename = include.contains(INCLUDE_FILENAME);
@@ -581,7 +583,7 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
       WritableMap edge = new WritableNativeMap();
       WritableMap node = new WritableNativeMap();
       boolean imageInfoSuccess =
-              putImageInfo(resolver, media, node, widthIndex, heightIndex, sizeIndex, dataIndex,
+              putImageInfo(resolver, media, node, widthIndex, heightIndex, orientationIndex, sizeIndex, dataIndex,
                       mimeTypeIndex, includeFilename, includeFileSize, includeFileExtension, includeImageSize,
                       includePlayableDuration);
       if (imageInfoSuccess) {
@@ -629,6 +631,7 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
           WritableMap node,
           int widthIndex,
           int heightIndex,
+          int orientationIndex,
           int sizeIndex,
           int dataIndex,
           int mimeTypeIndex,
@@ -643,7 +646,7 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
     String mimeType = media.getString(mimeTypeIndex);
 
     boolean isVideo = mimeType != null && mimeType.startsWith("video");
-    boolean putImageSizeSuccess = putImageSize(resolver, media, image, widthIndex, heightIndex,
+    boolean putImageSizeSuccess = putImageSize(resolver, media, image, widthIndex, heightIndex, orientationIndex,
             photoUri, isVideo, includeImageSize);
     boolean putPlayableDurationSuccess = putPlayableDuration(resolver, image, photoUri, isVideo,
             includePlayableDuration);
@@ -743,6 +746,7 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
           WritableMap image,
           int widthIndex,
           int heightIndex,
+          int orientationIndex,
           Uri photoUri,
           boolean isVideo,
           boolean includeImageSize) {
@@ -757,6 +761,13 @@ public class CameraRollModule extends ReactContextBaseJavaModule {
 
     int width = media.getInt(widthIndex);
     int height = media.getInt(heightIndex);
+    int orientation = media.getInt(orientationIndex);
+
+    if (isVideo && orientation > 0 && orientation != 0 && orientation != 180) {
+      int tempWidth = width;
+      width = height;
+      height = tempWidth;
+    }
 
     /* If the columns don't contain the size information, read the media file */
     if (width <= 0 || height <= 0) {
