diff --git a/node_modules/@react-native-camera-roll/camera-roll/android/build.gradle b/node_modules/@react-native-camera-roll/camera-roll/android/build.gradle
index 89d84b7..139313c 100644
--- a/node_modules/@react-native-camera-roll/camera-roll/android/build.gradle
+++ b/node_modules/@react-native-camera-roll/camera-roll/android/build.gradle
@@ -41,8 +41,19 @@ def getExtOrIntegerDefault(name) {
 
 apply plugin: 'com.android.library'
 
+def safeAppExtGet(prop, fallback) {
+  if (findProject(':app') == null) {
+      return fallback
+  }
+  def appProject = project(':app')
+  appProject?.ext?.has(prop) ? appProject.ext.get(prop) : fallback
+}
 
 def resolveReactNativeDirectory() {
+    def reactNativeLocation = safeAppExtGet("REACT_NATIVE_NODE_MODULES_DIR", null)
+    if (reactNativeLocation != null) {
+        return file(reactNativeLocation)
+    }
     // monorepo workaround
     // react-native can be hoisted or in project's own node_modules
     def reactNativeFromProjectNodeModules = file("${rootProject.projectDir}/../node_modules/react-native")
