diff --git a/node_modules/drizzle-kit/bin.cjs b/node_modules/drizzle-kit/bin.cjs
index 0edfed8..9fd74bc 100755
--- a/node_modules/drizzle-kit/bin.cjs
+++ b/node_modules/drizzle-kit/bin.cjs
@@ -4564,6 +4564,7 @@ var init_mysqlSchema = __esm({
       name: stringType(),
       columns: stringType().array(),
       isUnique: booleanType(),
+      isFulltext: booleanType().optional(),
       using: enumType(["btree", "hash"]).optional(),
       algorithm: enumType(["default", "inplace", "copy"]).optional(),
       lock: enumType(["default", "none", "shared", "exclusive"]).optional()
@@ -4689,14 +4690,15 @@ var init_mysqlSchema = __esm({
     MySqlSquasher = {
       squashIdx: (idx) => {
         index.parse(idx);
-        return `${idx.name};${idx.columns.join(",")};${idx.isUnique};${idx.using ?? ""};${idx.algorithm ?? ""};${idx.lock ?? ""}`;
+        return `${idx.name};${idx.columns.join(",")};${idx.isUnique};${idx.isFulltext};${idx.using ?? ""};${idx.algorithm ?? ""};${idx.lock ?? ""}`;
       },
       unsquashIdx: (input) => {
-        const [name, columnsString, isUnique, using, algorithm, lock] = input.split(";");
+        const [name, columnsString, isUnique, isFulltext, using, algorithm, lock] = input.split(";");
         const destructed = {
           name,
           columns: columnsString.split(","),
           isUnique: isUnique === "true",
+          isFulltext: isFulltext === "true",
           using: using ? using : void 0,
           algorithm: algorithm ? algorithm : void 0,
           lock: lock ? lock : void 0
@@ -11829,6 +11831,7 @@ We have encountered a collision between the index name on columns ${source_defau
             name,
             columns: indexColumns,
             isUnique: value.config.unique ?? false,
+            isFulltext: value.config.fulltext ? value.config.fulltext : undefined,
             using: value.config.using,
             algorithm: value.config.algorythm,
             lock: value.config.lock
@@ -11972,7 +11975,7 @@ We have encountered a collision between the index name on columns ${source_defau
       const tablePks = await db.execute(
         `SELECT table_name, column_name, ordinal_position
   FROM information_schema.table_constraints t
-  LEFT JOIN information_schema.key_column_usage k
+  INNER JOIN information_schema.key_column_usage k
   USING(constraint_name,table_schema,table_name)
   WHERE t.constraint_type='PRIMARY KEY'
       and table_name != '__drizzle_migrations'
@@ -12068,6 +12071,7 @@ We have encountered a collision between the index name on columns ${source_defau
         const constraintName = idxRow["INDEX_NAME"];
         const columnName = idxRow["COLUMN_NAME"];
         const isUnique = idxRow["NON_UNIQUE"] === 0;
+        const isFulltext = idxRow["INDEX_TYPE"] === "FULLTEXT" ? true : undefined;
         const tableInResult = result[tableName];
         if (typeof tableInResult === "undefined")
           continue;
@@ -12094,7 +12098,8 @@ We have encountered a collision between the index name on columns ${source_defau
               tableInResult.indexes[constraintName] = {
                 name: constraintName,
                 columns: [columnName],
-                isUnique
+                isUnique,
+                isFulltext
               };
             }
           }
@@ -13145,9 +13150,9 @@ The unique constraint ${source_default.underline.blue(
         if (value.length > 1) {
           value.sort();
           result[key].compositePrimaryKeys = {
-            [`${key}_${value.join("_")}_pk`]: {
+            [`${key}_${value.join("_")}`]: {
               columns: value,
-              name: `${key}_${value.join("_")}_pk`
+              name: `${key}_${value.join("_")}`
             }
           };
         } else if (value.length === 1) {
@@ -15366,7 +15371,7 @@ var init_sqlgenerator = __esm({
           const notNullStatement = column7.notNull ? " NOT NULL" : "";
           const defaultStatement = column7.default !== void 0 ? ` DEFAULT ${column7.default}` : "";
           const uniqueConstraint4 = column7.isUnique ? ` CONSTRAINT "${column7.uniqueName}" UNIQUE${column7.nullsNotDistinct ? " NULLS NOT DISTINCT" : ""}` : "";
-          const type = isPgNativeType(column7.type) ? column7.type : `"${column7.type}"`;
+          const type = isPgNativeType(alter_table_alter_column_set_type) ? column7.type : `"${column7.type}"`;
           statement += `	"${column7.name}" ${type}${primaryKeyStatement}${defaultStatement}${notNullStatement}${uniqueConstraint4}`;
           statement += i === columns.length - 1 ? "" : ",\n";
         }
@@ -16324,10 +16329,10 @@ ${BREAKPOINT}ALTER TABLE ${tableNameWithSchema} ADD CONSTRAINT ${statement.newCo
         return statement.type === "create_index" && dialect6 === "mysql";
       }
       convert(statement) {
-        const { name, columns, isUnique } = MySqlSquasher.unsquashIdx(
+        const { name, columns, isUnique, isFulltext } = MySqlSquasher.unsquashIdx(
           statement.data
         );
-        const indexPart = isUnique ? "UNIQUE INDEX" : "INDEX";
+        const indexPart = isFulltext ? "FULLTEXT INDEX" : isUnique ? "UNIQUE INDEX" : "INDEX";
         const value = columns.map((it) => `\`${it}\``).join(",");
         return `CREATE ${indexPart} \`${name}\` ON \`${statement.tableName}\` (${value});`;
       }
@@ -16854,7 +16859,7 @@ var init_jsonStatements = __esm({
             schema: schema4
           });
         }
-        if (((_d = column7.type) == null ? void 0 : _d.type) === "changed") {
+        if (((_d = column7.type) == null ? void 0 : _d.type) === "changed" && (column7.type.old !== "tinyint" && column7.type.new !== "boolean")) {
           statements.push({
             type: "alter_table_alter_column_set_type",
             tableName,
@@ -16870,6 +16875,7 @@ var init_jsonStatements = __esm({
           });
         }
         if (((_e = column7.primaryKey) == null ? void 0 : _e.type) === "deleted" || ((_f = column7.primaryKey) == null ? void 0 : _f.type) === "changed" && !column7.primaryKey.new && typeof compositePk === "undefined") {
+          
           dropPkStatements.push({
             ////
             type: "alter_table_alter_column_drop_pk",
@@ -16892,7 +16898,7 @@ var init_jsonStatements = __esm({
             columnPk
           });
         }
-        if (((_h = column7.default) == null ? void 0 : _h.type) === "changed") {
+        if (((_h = column7.default) == null ? void 0 : _h.type) === "changed" && (column7.type?.old !== "tinyint" || column7.default.new != column7.default.old)) {
           statements.push({
             type: "alter_table_alter_column_set_default",
             tableName,
