diff --git a/node_modules/drizzle-orm/mysql-core/indexes.cjs b/node_modules/drizzle-orm/mysql-core/indexes.cjs
index bef9aff..9a0e598 100644
--- a/node_modules/drizzle-orm/mysql-core/indexes.cjs
+++ b/node_modules/drizzle-orm/mysql-core/indexes.cjs
@@ -22,29 +22,32 @@ __export(indexes_exports, {
   IndexBuilder: () => IndexBuilder,
   IndexBuilderOn: () => IndexBuilderOn,
   index: () => index,
-  uniqueIndex: () => uniqueIndex
+  uniqueIndex: () => uniqueIndex,
+  fulltextIndex: () => fulltextIndex
 });
 module.exports = __toCommonJS(indexes_exports);
 var import_entity = require("../entity.cjs");
 class IndexBuilderOn {
-  constructor(name, unique) {
+  constructor(name, unique, fulltext) {
     this.name = name;
     this.unique = unique;
+    this.fulltext = fulltext;
   }
   static [import_entity.entityKind] = "MySqlIndexBuilderOn";
   on(...columns) {
-    return new IndexBuilder(this.name, columns, this.unique);
+    return new IndexBuilder(this.name, columns, this.unique, this.fulltext);
   }
 }
 class IndexBuilder {
   static [import_entity.entityKind] = "MySqlIndexBuilder";
   /** @internal */
   config;
-  constructor(name, columns, unique) {
+  constructor(name, columns, unique, fulltext) {
     this.config = {
       name,
       columns,
-      unique
+      unique,
+      fulltext
     };
   }
   using(using) {
@@ -77,12 +80,16 @@ function index(name) {
 function uniqueIndex(name) {
   return new IndexBuilderOn(name, true);
 }
+function fulltextIndex(name) {
+  return new IndexBuilderOn(name, false, true);
+}
 // Annotate the CommonJS export names for ESM import in node:
 0 && (module.exports = {
   Index,
   IndexBuilder,
   IndexBuilderOn,
   index,
-  uniqueIndex
+  uniqueIndex,
+  fulltextIndex
 });
 //# sourceMappingURL=indexes.cjs.map
\ No newline at end of file
diff --git a/node_modules/drizzle-orm/mysql-core/indexes.d.ts b/node_modules/drizzle-orm/mysql-core/indexes.d.ts
index 987a8e0..9069437 100644
--- a/node_modules/drizzle-orm/mysql-core/indexes.d.ts
+++ b/node_modules/drizzle-orm/mysql-core/indexes.d.ts
@@ -56,4 +56,5 @@ export type GetColumnsTableName<TColumns> = TColumns extends AnyMySqlColumn<{
 }>[] ? TTableName : never;
 export declare function index(name: string): IndexBuilderOn;
 export declare function uniqueIndex(name: string): IndexBuilderOn;
+export declare function fulltextIndex(name: string): IndexBuilderOn;
 export {};
diff --git a/node_modules/drizzle-orm/mysql-core/indexes.js b/node_modules/drizzle-orm/mysql-core/indexes.js
index 18df3c9..2a689b4 100644
--- a/node_modules/drizzle-orm/mysql-core/indexes.js
+++ b/node_modules/drizzle-orm/mysql-core/indexes.js
@@ -1,23 +1,25 @@
 import { entityKind } from "../entity.js";
 class IndexBuilderOn {
-  constructor(name, unique) {
+  constructor(name, unique, fulltext) {
     this.name = name;
     this.unique = unique;
+    this.fulltext = fulltext;
   }
   static [entityKind] = "MySqlIndexBuilderOn";
   on(...columns) {
-    return new IndexBuilder(this.name, columns, this.unique);
+    return new IndexBuilder(this.name, columns, this.unique, this.fulltext);
   }
 }
 class IndexBuilder {
   static [entityKind] = "MySqlIndexBuilder";
   /** @internal */
   config;
-  constructor(name, columns, unique) {
+  constructor(name, columns, unique, fulltext) {
     this.config = {
       name,
       columns,
-      unique
+      unique,
+      fulltext,
     };
   }
   using(using) {
@@ -50,11 +52,15 @@ function index(name) {
 function uniqueIndex(name) {
   return new IndexBuilderOn(name, true);
 }
+function fulltextIndex(name) {
+  return new IndexBuilderOn(name, false, true);
+}
 export {
   Index,
   IndexBuilder,
   IndexBuilderOn,
   index,
-  uniqueIndex
+  uniqueIndex,
+  fulltextIndex,
 };
 //# sourceMappingURL=indexes.js.map
\ No newline at end of file
diff --git a/node_modules/drizzle-orm/mysql-core/primary-keys.cjs b/node_modules/drizzle-orm/mysql-core/primary-keys.cjs
index 2da5d87..9a9e7fc 100644
--- a/node_modules/drizzle-orm/mysql-core/primary-keys.cjs
+++ b/node_modules/drizzle-orm/mysql-core/primary-keys.cjs
@@ -56,7 +56,7 @@ class PrimaryKey {
   columns;
   name;
   getName() {
-    return this.name ?? `${this.table[import_table.MySqlTable.Symbol.Name]}_${this.columns.map((column) => column.name).join("_")}_pk`;
+    return this.name ?? `${this.table[import_table.MySqlTable.Symbol.Name]}_${this.columns.map((column) => column.name).join("_")}`;
   }
 }
 // Annotate the CommonJS export names for ESM import in node:
