diff --git a/node_modules/next/dist/lib/typescript/writeConfigurationDefaults.js b/node_modules/next/dist/lib/typescript/writeConfigurationDefaults.js
index c80e26f..c22bdcd 100644
--- a/node_modules/next/dist/lib/typescript/writeConfigurationDefaults.js
+++ b/node_modules/next/dist/lib/typescript/writeConfigurationDefaults.js
@@ -117,6 +117,8 @@ function getDesiredCompilerOptions(ts) {
                 (ts.ModuleResolutionKind).Node12,
                 ts.ModuleResolutionKind.Node16,
                 ts.ModuleResolutionKind.NodeNext, 
+                // TODO remove once TypeScript 5.0 is supported natively by Next.js
+                ts.ModuleResolutionKind.Bundler, 
             ].filter((val)=>typeof val !== "undefined"),
             value: "node",
             reason: "to match webpack resolution"
@@ -228,10 +230,11 @@ async function writeConfigurationDefaults(ts, tsConfigPath, isFirstTimeSetup, is
             }
             // If `strict` is set to `false` or `strictNullChecks` is set to `false`,
             // then set `strictNullChecks` to `true`.
-            if (hasPagesDir && isAppDirEnabled && !userTsConfig.compilerOptions.strict && !("strictNullChecks" in userTsConfig.compilerOptions)) {
-                userTsConfig.compilerOptions.strictNullChecks = true;
-                suggestedActions.push(_chalk.default.cyan("strictNullChecks") + " was set to " + _chalk.default.bold(`true`));
-            }
+            // TODO seems to be a bug in next.js
+            // if (hasPagesDir && isAppDirEnabled && !userTsConfig.compilerOptions.strict && !("strictNullChecks" in userTsConfig.compilerOptions)) {
+            //     userTsConfig.compilerOptions.strictNullChecks = true;
+            //     suggestedActions.push(_chalk.default.cyan("strictNullChecks") + " was set to " + _chalk.default.bold(`true`));
+            // }
         }
     }
     if (!("exclude" in rawConfig)) {