diff --git a/node_modules/react-native/Libraries/Components/Pressable/Pressable.js b/node_modules/react-native/Libraries/Components/Pressable/Pressable.js
index 187b4c4..fbef285 100644
--- a/node_modules/react-native/Libraries/Components/Pressable/Pressable.js
+++ b/node_modules/react-native/Libraries/Components/Pressable/Pressable.js
@@ -334,8 +334,10 @@ function Pressable(props: Props, forwardedRef): React.Node {
   );
   const eventHandlers = usePressability(config);
 
+  const ViewComponent = props.ViewComponent ?? View;
+
   return (
-    <View
+    <ViewComponent
       {...restPropsWithDefaults}
       {...eventHandlers}
       ref={viewRef}
@@ -343,7 +345,7 @@ function Pressable(props: Props, forwardedRef): React.Node {
       collapsable={false}>
       {typeof children === 'function' ? children({pressed}) : children}
       {__DEV__ ? <PressabilityDebugView color="red" hitSlop={hitSlop} /> : null}
-    </View>
+    </ViewComponent>
   );
 }
 
