diff --git a/node_modules/react-native-photo-gallery-api/ios/RNPhotoGallery.m b/node_modules/react-native-photo-gallery-api/ios/RNPhotoGallery.m
index 59664e4..143a7d9 100644
--- a/node_modules/react-native-photo-gallery-api/ios/RNPhotoGallery.m
+++ b/node_modules/react-native-photo-gallery-api/ios/RNPhotoGallery.m
@@ -513,7 +513,7 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
         // convert to JPEG
         PHImageRequestOptions *const requestOptions = [PHImageRequestOptions new];
         requestOptions.networkAccessAllowed = YES;
-          requestOptions.deliveryMode = PHImageRequestOptionsDeliveryModeFastFormat;
+          requestOptions.deliveryMode = PHImageRequestOptionsDeliveryModeHighQualityFormat;
 
           CGSize const targetSize = CGSizeMake((CGFloat)asset.pixelWidth, (CGFloat)asset.pixelHeight);
           [[PHImageManager defaultManager] requestImageForAsset:asset
@@ -567,7 +567,14 @@ static void RCTResolvePromise(RCTPromiseResolveBlock resolve,
         editOptions.networkAccessAllowed = YES;
         
         [asset requestContentEditingInputWithOptions:editOptions completionHandler:^(PHContentEditingInput *contentEditingInput, NSDictionary *info) {
-          imageURL = contentEditingInput.fullSizeImageURL;
+          if (asset.mediaType == PHAssetMediaTypeImage) {
+            imageURL = contentEditingInput.fullSizeImageURL;
+          }
+          if (imageURL.absoluteString.length == 0 &&
+              [contentEditingInput.audiovisualAsset isKindOfClass:[AVURLAsset class]]) {
+            AVURLAsset *asset = (AVURLAsset *)contentEditingInput.audiovisualAsset;
+            imageURL = asset.URL;
+          }
           if (imageURL.absoluteString.length != 0) {
             
             filePath = [imageURL.absoluteString stringByReplacingOccurrencesOfString:@"pathfile:" withString:@"file:"];
